MapboxVision/src/main/java/com/mapbox/vision/utils/observable/CompositeListener.kt

```diff
@@ -0,0 +1,45 @@
+package com.mapbox.vision.utils.observable
+
+import com.mapbox.vision.mobile.core.utils.delegate.DelegateWeakRef
+import java.lang.ref.WeakReference
+import java.util.concurrent.CopyOnWriteArraySet
+
+interface CompositeListener<T> {
+    fun addListener(observer: T)
+    fun removeListener(observer: T)
+
+    open class Impl<T> : CompositeListener<T> {
```

Comment:
I'd suggest to show in name that listeners are held as weak refs, eg. `WeakReferenceImpl`